refactor: bigtable session separation with veneer wrappers

rahulKQL opened the pull request:
In progress.

rahulKQL committed:
refactor: add utility for converting Configuration to Bigtable Settings

rahulKQL committed:
In progress

rahulKQL committed:
In progress impl

rahulKQL committed:
More Progress

rahulKQL committed:
updated IBigtableSession with AutoClosable and finished it's implementation configuration.

igorbernstein2 commented on bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBigtableSession.java:
```diff
@@ -0,0 +1,23 @@
+
+  BulkRead createBulkRead(String tableId);
+
+  // TODO: Decide if this should be added or not. considering Instance op are deprecated.
```
I don't think this is necessary

igorbernstein2 commented on bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSessionClassicClient.java:
```diff
@@ -0,0 +1,589 @@
+import javax.net.ssl.SSLException;
+
+@InternalApi
+public class BigtableSessionClassicClient implements IBigtableSession {
```
Why are we duplicating BigtableSession? Can't we just delegate to it?

igorbernstein2 commented on bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java:
```diff
@@ -24,22 +24,41 @@
 import static com.google.cloud.bigtable.config.CallOptionsConfig.SHORT_TIMEOUT_MS_DEFAULT;
```
Can we split this class into classic & veneer variants? The only thing they need to share are the string constants for the keys, which can be extracted to a common class.
It'll make it easier to compare the 2 implementations

rahulKQL committed:
In progress

- delegated to  BigtableSession through BigtableSessionClassicClient when user choose existing impl.
- copied over Veneer impl of BigtableSession logic to BigtableSessionGCJClient.
